views/js/runner/provider/qti.js

```diff
@@ -569,14 +586,22 @@ define([
                         probeOverseer.stop();
                     });
             }
-
-            // prevent the item to be displayed while test runner is destroying
-            if (this.itemRunner) {
-                this.itemRunner.clear();
-            }
-            this.itemRunner = null;
         }
     };
 
+    /**
+     * Displays an exit message, then leaves the runner once the user has acknowledged
+     * @param {runner} runner
+     * @param {Object} data
+     */
+    function leaveRunner(runner, data) {
```

Comment:
The leave flow is incorrect, and it will be difficult to change. What about the following refactoring : 

Inside the `init` method : 
```js
this.on('leave', function(){
      this.trigger('destroy');
});
```

inside a plugin for example : 
```js
runner.before('leave', function(e, data){
  var done = e.done();
    var context = runner.getTestContext();
       runner.disableItem(context.itemUri);
        runner.trigger('alert', data.message, function () {
                runner.trigger('endsession', 'teststate', data.code);
                done();
        });
});
```